models/shopcart.py

```diff
@@ -39,3 +39,12 @@ def all():
     def remove_all():
         """ Remove all Shopcarts from memory """
         Shopcart.__data = []
+    
+    @staticmethod
+    def find(uid):
+        """ Find a Shopcart by it's uid"""
+        for cart in Shopcart.__data:
+            if cart.uid == uid:
+                return cart
+
+        return None
```

Comment:
Should we maybe create a new Error class, like `NotFoundError`, and raise that instead of returning None?

Reply:
We only need an error if we to modifying a shopcart that does not exist so an error is not necessary in this case in my opinion.

Reply:
I think the API should handle 404 not found in case they call a find on a non-existant cart.